tests/ResourceSyncingTest.php

```diff
@@ -763,6 +763,45 @@ function creatingResourceInTenantDatabaseCreatesAndMapInCentralDatabase()
     expect(ResourceUser::first()->role)->toBe('commenter');
 }
 
+test('resources are synced only when sync is enabled', function () {
+    [$tenant1, $tenant2] = createTenantsAndRunMigrations();
+    migrateUsersTableForTenants();
+
+    tenancy()->initialize($tenant1);
+
+    // Tenant model returns false from `shouldSync`
+    // central model will not be created
+    TenantUserWithDisabledSync::create([
+        'global_id' => 'absd',
+        'name' => 'John Doe',
+        'email' => 'john@localhost',
+        'password' => 'password',
+        'role' => 'commenter',
+    ]);
+
+    tenancy()->end();
+
+    expect(CentralUser::all())->toHaveCount(0);
+    expect(CentralUser::whereGlobalId('absd')->first())->toBeNull();
+
+    $centralUser = CentralUserWithDisabledSync::create([
+        'global_id' => 'acme',
+        'name' => 'John Doe',
+        'email' => 'john@localhost',
+        'password' => 'password',
+        'role' => 'commenter',
+    ]);
+
+    // central model returns false from `shouldSync`
+    // resource model will not be created
+    $centralUser->tenants()->attach('t2');
+
+    $tenant2->run(function () {
+        expect(ResourceUser::all())->toHaveCount(0);
+        expect(ResourceUser::whereGlobalId('acme')->first())->toBeNull();
```

Comment:
I think it doesn't make sense to be interacting with `ResourceUser` and `CentralUser` in this test. You defined new models, so you should be using exclusively those. Or is there any reason to use these specifically?

(I just want an answer here, not code changes, I'm working on rewriting the tests myself)

Reply:
> You defined new models, so you should be using exclusively those. Or is there any reason to use these specifically?

I'm using them for `creation` only.  There is no specific reason. We can use newly defined models here too.